File: 1019-squares-of-a-sorted-array/squares-of-a-sorted-array.py
class Solution:
    def sortedSquares(self, nums: List[int]) -> List[int]:
        l = 0
        r = len(nums)-1
        n = len(nums)
        res = [0]*n
        for i in range(n-1, -1, -1):
            if abs(nums[l])<abs(nums[r]):
                sq = abs(nums[r])
                r-=1
            else:
                sq = abs(nums[l])
                l+=1
            res[i]=sq*sq
        return res

File: 1019-squares-of-a-sorted-array/squares-of-a-sorted-array.java
class Solution {
    public int[] sortedSquares(int[] nums) {
        int n = nums.length;
        int[] res = new int[n];
        int left = 0;
        int right = n-1;
        int sq;

        for (int i=n-1; i>=0; i--){
            if (Math.abs(nums[right])>Math.abs(nums[left])){
                sq = nums[right];
                right--;
            }
            else {
                sq = nums[left];
                left++;
            }
            res[i]=sq*sq;
        }
        return res;

    }
}